editor/src/components/canvas/design-panel-root.tsx

```diff
@@ -149,42 +151,80 @@ export const DesignPanelRoot = betterReactMemo('DesignPanelRoot', (props: Design
-          enable={{
-            top: false,
-            right: props.isUiJsFileOpen,
-            bottom: false,
-            topRight: false,
-            bottomRight: false,
-            bottomLeft: false,
-            topLeft: false,
-          }}
-          className='resizableFlexColumnCanvasCode'
-          style={{
-            ...UtopiaStyles.flexColumn,
-            display: !props.isUiJsFileOpen || interfaceDesigner.codePaneVisible ? 'flex' : 'none',
-            width: interfaceDesigner.codePaneWidth,
-            height: '100%',
-            position: 'relative',
-            overflow: 'hidden',
-            justifyContent: 'stretch',
-            alignItems: 'stretch',
-            borderLeft: `1px solid ${UtopiaTheme.color.subduedBorder.value}`,
-          }}
-        >
-          <CodeEditorWrapper />
-        </Resizable>
-        {props.isUiJsFileOpen ? <CanvasWrapperComponent {...props} /> : null}
+        <SimpleFlexColumn style={{ flexGrow: props.isUiJsFileOpen ? undefined : 1 }}>
+          <SimpleFlexRow
+            className='tabRail'
+            style={{
+              minHeight: 30,
+              height: 30,
+              borderBottom: `1px solid ${UtopiaTheme.color.subduedBorder.value}`,
+              alignItems: 'stretch',
+              justifyContent: 'stretch',
+              backgroundColor: 'transparent',
+              overflowX: 'hidden',
+            }}
+          >
+            <FileTabs />
+          </SimpleFlexRow>
+          <Resizable
+            defaultSize={{ width: interfaceDesigner.codePaneWidth, height: '100%' }}
+            size={{
+              width: props.isUiJsFileOpen ? interfaceDesigner.codePaneWidth : '100%',
+              height: '100%',
+            }}
+            onResizeStop={onResizeStop}
+            onResize={onResize}
+            enable={{
+              top: false,
+              right: props.isUiJsFileOpen,
+              bottom: false,
+              topRight: false,
+              bottomRight: false,
+              bottomLeft: false,
+              topLeft: false,
+            }}
+            className='resizableFlexColumnCanvasCode'
+            style={{
+              ...UtopiaStyles.flexColumn,
+              display: !props.isUiJsFileOpen || interfaceDesigner.codePaneVisible ? 'flex' : 'none',
+              width: interfaceDesigner.codePaneWidth,
+              height: '100%',
+              position: 'relative',
+              overflow: 'hidden',
+              justifyContent: 'stretch',
+              alignItems: 'stretch',
+              borderLeft: `1px solid ${UtopiaTheme.color.subduedBorder.value}`,
+            }}
+          >
+            <CodeEditorWrapper />
+          </Resizable>
+        </SimpleFlexColumn>
+        {props.isUiJsFileOpen ? (
+          <SimpleFlexColumn style={{ flexGrow: 1 }}>
+            <SimpleFlexRow
+              className='topMenu'
+              style={{
+                minHeight: 30,
+                height: 30,
```

Comment:
could you make this height a const here, and use it for the NavigatorComponent's top too?

Reply:
fixed